app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -10,9 +10,8 @@ def command
 
   def registration_args
     args = registration_params.except(*ignored_query_args)
-
-    args['setup_insights'] = setup_insights.to_s
-    args['setup_remote_execution'] = setup_remote_execution.to_s
+    args['setup_insights'] = registration_params['setup_insights']
```

Comment:
Don't you want to use `ActiveRecord::Type::Boolean.new.deserialize` here? It seems like it was used before and also is being used in `Registration` contorller concern.

Reply:
I don't want to, because with `Boolean.new.deserialize`  auto-selecting option in `options_for_select` for `<select>` elements [0] stop working.

[0] https://github.com/theforeman/foreman/pull/8342/files#diff-11618cf852993f247cc0bbc117733623ce4c6bee87d332ef8232fd103a3fac90R61 & https://github.com/theforeman/foreman/pull/8342/files#diff-11618cf852993f247cc0bbc117733623ce4c6bee87d332ef8232fd103a3fac90R78